core/autoscaling_test.go

```diff
@@ -2674,3 +2674,68 @@ func TestReplaceOnDemandInstanceWithSpot(t *testing.T) {
 		})
 	}
 }
+
+func TestGetInstanceTypeByTagInASG(t *testing.T) {
+	tests := []struct {
+		name     string
+		asgTags  []*autoscaling.TagDescription
+		tagKey   string
+		expected *string
```

Comment:
The prototype of `getInstanceTypeByTagInASG` is : `func (a *autoScalingGroup) getInstanceTypeByTagInASG() (string, error) {` so I assume that the `error` field should also be tested. #